crypto.go

```diff
@@ -117,7 +117,13 @@ func generateCertificateVerify(handshakeBodies []byte, privateKey crypto.Private
 	switch p := privateKey.(type) {
 	case ed25519.PrivateKey:
 		// https://crypto.stackexchange.com/a/55483
-		return p.Sign(rand.Reader, hashed, crypto.Hash(0))
+		// [..] You do not have to hash the inputs to Ed25519 because hashing is already
+		// part of Ed25519 itself. If you do hash inputs in advance, you become vulnerable
+		// to collisions in the hash function you use. [..]
+		//
+		// ed25519.PrivateKey.Sign(): signs the given message with priv. Ed25519 performs
+		// two passes over messages to be signed and therefore cannot handle pre-hashed messages.
+		return p.Sign(rand.Reader, handshakeBodies, crypto.Hash(0))
```

Comment:
It looks better to skip https://github.com/pion/dtls/pull/470/files#diff-69c5c48c1da9bf0e59ac32c7d0c0089fee30329073ffcc86cf7200f224f44ecfR111-R115 which isn't used in this path

Reply:
GitHub is doing something wrong for me I think because that link ends up in the middle of `e2e_test.go` without highlighting anything. What do you want us to skip?

Reply:
https://github.com/pion/dtls/blob/8f9e598d8c5a129bc234f26abe7fdcf21b884626/crypto.go#L111-L115
`hashed` is not used in this `case`.

Reply:
Ah, right yes. Lets fix that!